Rework observability page titles and summaries

ahus1 opened the pull request:
Closes #38583

Before: 

![image](https://github.com/user-attachments/assets/05422b78-3001-4fee-941d-632c46ff3180)

After: 

![image](https://github.com/user-attachments/assets/01ff041b-354a-4a71-8be4-24538da82e7b)


<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

ahus1 committed:
Rework observability page titles and summaries

Closes #38583

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>

ahus1 commented:
@pruivo / @ryanemerson / @mhajas / @keycloak/cloud-native - happy to hear your comments on the new titles in the observability guide.

ahus1 commented:
@andymunro - I updated the title to the new pattern. Happy to get a high-level comment now if you think this is going into the right direction. 

I will ask for a full review of the change once I have the feedback from the other engineers.

ahus1 commented:
@ryanemerson - I would allow 5-7 words on a title to reveal more of the "what for" and "to what benefit" this guide is about. Some can be shortened when putting some more thoughts into it. Still I think "Enable Metrics" is too short as it doesn't give much insight from the title.

ahus1 commented:
@ryanemerson - adding to the above, "Enable metrics" might be the title for the one already knowing what they are searching for, the longer title might be for the one looking for an answer to a problem but they didn't know it would be metrics.

ryanemerson commented:
Isn't that the purpose of the description though? Using the metrics as an example in the new screenshot, it seems to me that the title and description overlap a lot

mhajas commented:
I agree with both here. It is now quite hard to come to the page and just find grafana dashboards because there is too much text. Maybe if we could somehow highlight what the guide is about. Something like this:
![image](https://github.com/user-attachments/assets/33031cea-bd98-41a0-83f0-3e1516bc498a)

Or maybe we could have some sort of tags in each tile, similarly to how github issue tags are displayed.

ryanemerson commented:
I think the highlighting of keywords helps a lot. I like the tagging idea as well

andymunro commented:
Thanks, @ahus1 for these clear improvements.  I like the new format. 
The highlighting suggested by @mhajas could work.
Another approach is to put the feature up front. For example

Using health checks to track status information of instances

ahus1 commented:
@andymunro - when I put the feature in front, I seem to end up with the verb "using" at the beginning in most cases, and sometimes "enabling". This isn't the strong verbs like "visualizing", "troubleshooting", "gaining insights" etc., which is IMHO a loss.

Can you help me out here?

andymunro commented:
Yes, I would go with your original wording. It works. The focus should be on the less experienced user. So I am not concerned that the expert user will have trouble finding the content.  However, you could add highlighting as suggested if you think it's needed.

ahus1 committed:
Shortening the titles to make them simpler to scan for readers

Closes #38583

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>

ahus1 commented:
After the feedback I've shortened the titles, and I hope this addresses most of the concerns from @ryanemerson and @mhajas.

Additional highlighting or hash tags would be out of scope for this PR as this is primarily addressing the issues we see in downstream docs which are very long titles and non-descriptive titles. 

Please let me know if you are good to proceed here.

vmuzikar reviewed (commented):
@ahus1 Thank you for the PR.

From my perspective, the original shorter titles were better. To me it feels easier to read 4 words as the title than a whole sentence, and it helps me to navigate the guides easier and more clearly. The updated summaries look good, though.

That said, I understand this is subjective, so I'm not vetoing it if others like the new longer titles.

ryanemerson reviewed (commented):
Thank you for shortening the titles, I think this is an improvement :+1: 

One observation though, I noticed that the other non-observability guides also follow the convention of having very succinct titles. Is there a plan to also change these so that we have a unified style for all guides?

mhajas commented on docs/guides/observability/grafana-dashboards.adoc:
```diff
@@ -2,13 +2,12 @@
 <@tmpl.guide
-title="{project_name} Grafana dashboards"
-summary="Learn how to import {project_name} provided Grafana dashboards to your Grafana instance" >
+title="Visualizing cluster activities with dashboards"
```
What about `Visualizing deployment activities` if we want to avoid word Keycloak there

ahus1 commented on docs/guides/observability/tracing.adoc:
```diff
@@ -4,14 +4,14 @@
 
-<@tmpl.guide title="Enabling Tracing"
-summary="Learn how to enable distributed tracing in {project_name}"
+<@tmpl.guide title="Tracking down root causes with tracing"
```
ok

ahus1 committed:
Rewording

Closes #38583

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>

mhajas commented on docs/guides/observability/keycloak-service-level-indicators.adoc:
```diff
@@ -2,8 +2,8 @@
 <@tmpl.guide
-title="{project_name} service level indicators (SLIs)"
-summary="Learn about the Service Level Indicators to monitor your {project_name} deployment's performance"
+title="Monitoring performance with Service Level Indicators"
```
I am not sure performance is the correct word here, as SLIs are not only about performance. It kind of represents overall correctness of the system. 

Would it make sense to change it something like `Detect misbehavior [of the system] using Service Level Indicators`?

ahus1 replied:
The guide is mostly about latency and error rates that Keycloak provides, and availability that we use an external tool for (Prometheus' own `up` metric). 

As it is more in the lines of "how well the system performs" and as I would prefer a positive statement over negative statement, I'd like to stay with the term.

mhajas reviewed (approved):
Thank you for the update. I really like the updated titles!

andymunro reviewed (approved):
Looks great, @ahus1 Thanks.
